### ISSUE TITLE
Timestamps should always be two or none

### ISSUE THREAD

**Issue by tomdemeyer** [histograph/core#60]
When only `validSince` is given, add  `validUntil` in the far future on import;
When only `validUntil` is given, add `validSince` in the far past on import.

**Comment by bertspaan** [comment]
I don't think I agree: seems like something you would do on application level, or in queries.

**Comment by tomdemeyer** [comment]
makes cypher queries very elaborate; I cannot see a use-case where you would want to have one but not the other..
… and if you need such a thing, ignoring the offending timestamp is easy if it is 0 or 10000000 (for instance)

## 

Tom Demeyer
Waag Society
http://waag.org

**Comment by bertspaan** [comment]
Ok, if you mean adding always two _timestamps_ for Neo4j PITs, then I agree.

**Comment by tomdemeyer** [comment]
Mmmm. Maybe.
If you do a query for concurrent events you’ll never want to see PITS without _any_ time info.
When one of the two is present, it signals time info was provided..
Of course all also possible in the api.. At the moment I ignore all pits with both NULLs very easily.

Hack the knot but through & let me know.

## 

Tom Demeyer
Waag Society
http://waag.org

**Comment by bertspaan** [comment]
No, I think it's a good idea, I didn't realize you meant the Neo4j-only timestamps!

**Comment by sbocconi** [comment]
Is this issue from Trello related:

## zoeken op before and after: plan maken voor huidige beperkingen

"ik wil inzoomen op een bounding box en dan met slider door de tijd kunnen lopen en zien wat er bij komt en af gaat."

"Open einde" records worden niet gevonden. Wanneer is dat een probleem en hoe lossen we dat dan op?
Straten met datum van aanleg worden niet teruggevonden (o.a. in De Rijp). Zoeken op meerdere dimensies die in verschillende datasets voorkomen. Bij geometrie in set 1 en tijd in set 2 levert als resultaat altijd '0'.

**Comment by tomdemeyer** [comment]
No, different issue..

## 

Tom Demeyer
Waag Society
http://waag.org
